src/dolphin/install/installation.ts

```diff
@@ -176,11 +176,16 @@ export class DolphinInstallation {
   }
 
   private async _isOutOfDate(latestVersion: string): Promise<boolean> {
+    const dolphinVersion = await this.getDolphinVersion();
+    return lt(dolphinVersion, latestVersion);
+  }
+
+  public async getDolphinVersion(): Promise<string> {
```

Comment:
```suggestion
  public async getDolphinVersion(): Promise<string | null> {
```
I think we should return null instead of an empty string if we can't find the dolphin executable.